src/components/Uploader.js

```diff
@@ -11,11 +12,22 @@ class Uploader extends Component {
 
   handleChangeImage = e => {
     const { target: { files } } = e;
-    const { getMedia } = this.props;
-    const filesToStore = [];
-    Array.from(files).forEach(file => filesToStore.push(file));
-
-    getMedia(filesToStore, this.fileUploader);
+    const { getMedia, acceptType, setSpinner } = this.props;
+    setSpinner(true);
+    if (acceptType === "images/*") {
+      Promise.all(Array.from(files).map(file => imageCompression(file, {maxSizeMB: .5})))
```

Comment:
Does this have an effect on image resolution?

Reply:
and we will compress all images irrespective of their sizes right?

Reply:
Yes. Is that ok?

Reply:
I do not have an extensive knowledge on image compression. Since the maximum size is 10MB, I guess compressing all images, irrespective of their sizes to .5MB is fine. I was wondering whether there is any advantage in compressing the images based on their sizes. Did you come across anything like that when you did the reading for image compression?

Reply:
As discussed offline, we can go forward with this logic for now. Thanks!